### PR TITLE
[Vue] migrate smaller core home common directives and deprecate others

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based off of #18445.**

Changes:
* Migrate dropdown-button.
* Migrate piwik-show-sensitive-data. (Unused, mostly for BC.)
* Mark other common directives as deprecated, they will be removed rather than migrated.

image

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
UI tests should be passsing for this one

**Review by peterhashair — COMMENTED** [review]
Review VUE parts, CSS, and interface working as expected. Failed seems unrelated. There are a couple of conflicts maybe need a fix.

File: plugins/CoreHome/angularjs/common/directives/show-sensitive-data.js [hunk 1]
```diff
@@ -5,17 +5,7 @@
...
- * On click on the element the full data will be shown
- *
- * Configuration attributes:
- * data-show-characters          number of characters to show in clear text (defaults to 6)
- * data-click-element-selector   selector for element that will show the full data on click (defaults to element)
- *
- * Example:
- * <div piwik-show-sensitive-date="some text"></div>
- */
+
```
Comment (sgiehl) [inline plugins/CoreHome/angularjs/common/directives/show-sensitive-data.js:8]: @diosmosis Is there any reason why this directive wasn't removed completely, even though there is an vue/angularjs adapter?

Comment (diosmosis) [reply plugins/CoreHome/angularjs/common/directives/show-sensitive-data.js:8]: no, will remove

**Pull request merged**